components/core/src/fs.rs

```diff
@@ -712,21 +746,14 @@ impl AtomicWriter {
         let tempfile = tempfile::NamedTempFile::new_in(parent)?;
         Ok(Self { dest: dest_path.to_path_buf(),
                   tempfile,
-                  permission: None })
+                  permissions: Permissions::default() })
     }
 
-    #[cfg(not(windows))]
-    /// with_permission sets the target mode for the destination path.
+    /// Sets the given permissions for the destination path.
     ///
-    /// note: The mode is set explicitly and not subject to the
-    /// process umask.
-    pub fn with_permissions(&mut self, mode: u32) { self.permission = Some(mode); }
-
-    #[cfg(windows)]
-    /// with_permission sets given permissions on the destination path.
-    pub fn with_permissions(&mut self, entries: Vec<win_perm::PermissionEntry>) {
-        self.permission = Some(entries);
-    }
+    /// Note: On Unix platforms, permissions are set explicitly and
+    /// not subject to the process umask.
+    pub fn with_permissions(&mut self, permissions: Permissions) { self.permissions = permissions; }
```

Comment:
What would you think about making this a constructor?

Reply:
Yup! Handled in https://github.com/habitat-sh/habitat/pull/7545/commits/a63b8a0577fcc27dea88b67befe917c51dfd790c